modules/architecture-machine-roles.adoc

```diff
@@ -4,61 +4,31 @@
+
+In {product-title}, machine sets control the worker machines. Machines with the worker role drive compute workloads that are governed by a specific machine pool that autoscales them. Because {product-title} has the capacity to support multiple machine types, the worker machines are classed as _compute_ machines. In this release, the terms _worker machine_ and _compute machine_ are used interchangeably because the only default type of compute machine is the worker machine. In future versions of {product-title}, different types of compute machines, such as infrastructure machines, might be used by default.
 
 [id="defining-masters_{context}"]
 == Cluster masters
 
-In a Kubernetes cluster, the master nodes run services that are required to
-control the Kubernetes cluster. In {product-title}, the master machines are
-the control plane. They contain more
-than just the Kubernetes services for managing the {product-title} cluster.
-Because all of the machines with the control plane role are master machines,
-the terms _master_ and _control plane_ are used interchangeably to describe
-them. Instead of being grouped into a
-machine set, master machines are defined by a series of standalone machine API
-resources. Extra controls apply to master machines to prevent you from deleting
-all master machines and breaking your cluster.
+In a Kubernetes cluster, the master nodes run services that are required to control the Kubernetes cluster. In {product-title}, the master machines are the control plane. They contain more than just the Kubernetes services for managing the {product-title} cluster. Because all of the machines with the control plane role are master machines, the terms _master_ and _control plane_ are used interchangeably to describe them. Instead of being grouped into a machine set, master machines are defined by a series of standalone machine API resources. Extra controls apply to master machines to prevent you from deleting all master machines and breaking your cluster.
 
 [NOTE]
 ====
-Use three master nodes. Although you can theoretically
-use any number of master nodes, the number is constrained by etcd quorum due to
-master static pods and etcd static pods working on the same hosts.
+Exactly three master nodes should be used for all production deployments. Any high availability requirements that exceed what is provided by three masters should instead be met through deployment of multiple {product-title} clusters, each with three masters.
```

Comment:
@sferich888  I updated the language here. PTAL.

Reply:
Do you really want to use "should" instead of "must?"

If we were going to add a note about the ability to use a worker-less cluster, this would be the place to put it, I think.

Reply:
@kalexand-rh Good catch!